File: ios/YamabikoChat/Networking/Models/LLMProvider.swift
import Foundation

enum LLMProvider: String, Codable, CaseIterable, Sendable {
    case gemini = "GEMINI"
    case openRouter = "OPENROUTER"
    case openCodeGo = "OPENCODE_GO"
    case clinePass = "CLINEPASS"
    case alibabaCodingPlan = "ALIBABA_CODING_PLAN"
    case openAI = "OPENAI"
    case openAICompat = "OPENAI_COMPAT"
    case miniMax = "MINIMAX"
    case codexAuth = "CODEX_AUTH"
    case superGrok = "SUPERGROK"
    case zai = "ZAI"
    case appleIntelligence = "APPLE_INTELLIGENCE"

    init(rawOrDefault value: String) {
        switch value.uppercased() {
        case "GEMINI_AUTH":
            self = .gemini
        case "QWEN_CODE":
            self = .openRouter
        default:
            self = LLMProvider(rawValue: value.uppercased()) ?? .gemini
        }
    }

    var supportsClientWebSearchTool: Bool {
        self != .appleIntelligence
    }
}

File: app/src/main/java/com/porarri/yamabikochat/data/model/LLMProvider.kt
package com.porarri.yamabikochat.data.model

enum class LLMProvider(val rawValue: String) {
    GEMINI("GEMINI"),
    OPENROUTER("OPENROUTER"),
    OPENCODE_GO("OPENCODE_GO"),
    CLINEPASS("CLINEPASS"),
    ALIBABA_CODING_PLAN("ALIBABA_CODING_PLAN"),
    OPENAI("OPENAI"),
    OPENAI_COMPAT("OPENAI_COMPAT"),
    MINIMAX("MINIMAX"),
    CODEX_AUTH("CODEX_AUTH"),
    SUPERGROK("SUPERGROK"),
    ZAI("ZAI"),
    APPLE_INTELLIGENCE("APPLE_INTELLIGENCE");

    val supportsClientWebSearchTool: Boolean
        get() = this != APPLE_INTELLIGENCE

    companion object {
        fun fromRawOrDefault(value: String): LLMProvider {
            return when (value.trim().uppercase()) {
                "GEMINI_AUTH" -> GEMINI
                "QWEN_CODE" -> OPENROUTER
                else -> entries.firstOrNull { it.rawValue == value.trim().uppercase() } ?: GEMINI
            }
        }

        fun fromRawOrNull(value: String): LLMProvider? {
            return when (value.trim().uppercase()) {
                "GEMINI_AUTH" -> GEMINI
                "QWEN_CODE" -> OPENROUTER
                else -> entries.firstOrNull { it.rawValue == value.trim().uppercase() }
            }
        }
    }
}
